langchain4j/src/main/java/dev/langchain4j/service/tool/ToolExecutionResult.java

```diff
@@ -56,14 +57,14 @@ public Object result() {
      * @see #result()
      */
     public String resultText() {
-        if (resultText == null && resultTextSupplier != null) {
-            synchronized (this) {
-                if (resultText == null) {
-                    resultText = resultTextSupplier.get();
-                }
-            }
+        String text = resultText.get();
+        if (text == null && resultTextSupplier != null) {
+            text = resultTextSupplier.get();
+            resultText.compareAndSet(null, text);
+            // Re-read to ensure we return the winning value in case of race
+            text = resultText.get();
         }
-        return resultText;
+        return text;
```

Comment:
I believe that this whole method could be implemented with a one-liner as it follows (please double check):

```
  public String resultText() {
      return resultText.updateAndGet(text -> text != null ? text : resultTextSupplier.get());
  }
```

Reply:
Agreed, I have done the changes